guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/list/RepositoryItemPresenter.java

```diff
@@ -71,6 +71,11 @@ public void setRepository( final Repository repository,
         view.refresh();
     }
 
+    public void refreshBranches() {
+        populateBranches( view.getSelectedBranch() );
+        view.refresh();
```

Comment:
Minor comment; ```setSelectedBranch(..)``` in ```populateBranches(..)``` calls ```branchesDropdown.refresh()``` which is the same as ```view.refresh()``` so the call here should not be strictly needed; but since it's not an expensive operation it probably does no harm.